lib/chef_core/text.rb

```diff
@@ -23,35 +23,57 @@
 # down the road and don't want to have to change all of our commands.
 module ChefCore
   module Text
-    def self._error_table
-      # Though ther may be several translations, en.yml will be the only one with
-      # error metadata.
-      path = File.join(_translation_path, "errors", "en.yml")
-      raw_yaml = File.read(path)
-      @error_table ||= YAML.load(raw_yaml, _translation_path, symbolize_names: true)[:errors]
+
+    DEFAULT_LOCALIZATION_PATH = File.absolute_path(File.join(File.dirname(__FILE__), "..", "..", "i18n"))
+
+    # Set up this gem's localization as the only one present
+    def self.reset!
+      @localization_paths = []
+      @raw_localization_paths = []
+      add_localization(DEFAULT_LOCALIZATION_PATH)
+    end
+
+    def self.add_localization(base_path)
+      return if @raw_localization_paths.include? base_path
+      # @localization_paths will get modified by R18n, so we'll
+      # keep them as strings as well, to ensure we can avoid duplicate loading.
+      errors_path = File.join(base_path, "errors")
+      @localization_paths << base_path
+      @localization_paths << errors_path
+      @raw_localization_paths << base_path
+      @raw_localization_paths << errors_path
+      reload!
     end
 
-    def self._translation_path
-      @translation_path ||= File.join(File.dirname(__FILE__), "..", "..", "i18n")
+    def self.add_gem_localization(gem_name)
+      spec = Gem::Specification.find_by_name(gem_name)
+      path = File.join(spec.gem_dir, "i18n")
+      if File.directory? path
+        add_localization(path)
+      end
     end
 
-    # TODO BOOTSTRAP - must support loading from each gem's
-    #                  text (in particular i18n/errors/*.yml)
-    #                  We may need to accept the translation paths
-    #                  as a list.
-    def self.load
-      R18n.from_env(Text._translation_path)
-      R18n.extension_places << File.join(Text._translation_path, "errors")
+    def self.reload!
```

Comment:
TODO: ensure error message translations only exist above chef-core; look at ensuring messages, etc can't be overwritten?